### PR TITLE
[ONNX] Use fixed test input for flaky test

### LINKED ISSUE #74501: test/onnx/test_pytorch_onnx_onnxruntime.py::TestONNXRuntime_opset12::test_quantized_linear is flaky
Example failures:
- https://github.com/pytorch/pytorch/runs/5593111469?check_suite_focus=true
- https://github.com/pytorch/pytorch/runs/5635175017?check_suite_focus=true
- https://github.com/pytorch/pytorch/runs/5634164753?check_suite_focus=true

### PR DESCRIPTION
**Pull request by BowenBao**
Fixes #74501

Random weight and input sometimes result in slight output mismatch
between PyTorch and ONNX model. This is expected as the ONNX
model is executed in QDQ format.

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -10546,13 +10546,18 @@ def forward(self, x):
     #       such that inputs and outputs do not always overflow/underflow.
     #       Otherwise test results could be inaccurate.
     @skipIfUnsupportedMinOpsetVersion(10)
-    @unittest.skip("https://github.com/pytorch/pytorch/issues/74501")
     def test_quantized_linear(self):
         model = torch.nn.quantized.Linear(4, 8)
+        # Set fixed weight to avoid flaky test.
+        weight = torch.quantize_per_tensor(
+            torch.arange(32).view(8, 4).to(torch.float),
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:10553]: Is `.to(torch.float)` needed? I thought that was the default for these tests?

if it is needed, wouldn't setting `dtype=torch.float` in the call to `arange` be more efficient?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -10546,13 +10546,18 @@ def forward(self, x):
...
+            0.5, 0, torch.qint8)
         # Set non-zero bias.
         bias = torch.arange(8).to(torch.float)
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:10556]: same comments about `.to(torch.float)`

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 3]
```diff
@@ -10546,13 +10546,18 @@ def forward(self, x):
...
-        model.set_weight_bias(model.weight(), bias)
+        model.set_weight_bias(weight, bias)
+        # Set fixed input to avoid flaky test.
         input = torch.randn(4, 4)
+        input = torch.arange(16).view(4, 4).to(torch.float) - 8
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:10560]: same comments about `.to(torch.float)`

**Review by garymm — APPROVED** [review]
Thanks for fixing!

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 1 above)
Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:10553]: yes it's needed, arange default returns int64. I will update with the dtype suggestion.

**Pull request closed without merging**